### PR TITLE
SNOW-870432: use_logical_type for inferring timezone in pandas df

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #870432, #991

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Port the fix from snowflake python connector to use_logical_type when inferring datatype from parquet files.

### CODE REVIEW

File: tests/integ/test_dataframe.py [hunk 1]
```diff
@@ -1478,6 +1479,41 @@ def test_create_dataframe_with_semi_structured_data_types(session):
     )
 
 
+@pytest.mark.skipif(not is_pandas_available, reason="pandas is required")
+def test_create_dataframe_with_pandas_df(session):
```
Comment (sfc-gh-sfan) [inline tests/integ/test_dataframe.py:1483]: Do we need to port some changes to sp connector too to support `use_logical_type` in stored proc?

Comment (sfc-gh-aalam) [reply tests/integ/test_dataframe.py:1483]: That's right. We need to port this to sproc too

Comment (sfc-gh-aalam) [reply tests/integ/test_dataframe.py:1483]: https://github.com/snowflakedb/Stored-Proc-Python-Connector/pull/140

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -2008,6 +2014,7 @@ def create_dataframe(
                 quote_identifiers=True,
                 auto_create_table=True,
                 table_type="temporary",
+                use_logical_type=True,
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:2017]: Is this a behavior change (i.e. wrong behavior -> correct behavior)? Should we highlight the implication in change log?

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/session.py:2017]: I'll update the changelog detailing what exactly is going to change

**Review by sfc-gh-sfan — APPROVED** [review]

**Review by sfc-gh-aling — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,12 @@
 
 - Fixed a bug in `DataFrame.na.fill` that caused Boolean values to erroneously override integer values.
 - Fixed sql simplifier for filter with window function columns in select.
+- Fixed a bug in `Session.create_dataframe` where the snowpark dataframes created using pandas dataframes were not inferring the type for timestamp columns correctly. The behavior is as follows:
+  - Earlier timestamp columns without a timezone would be inferred as `LongType()` but will now be correctly inferred as `TimestampType(TimestampTimeZone.NTZ)`.
+  - Earlier timestamp columns without a timezone would be converted to nanosecond epochs, but will now be correctly be maintained as timestamp values.
```
Comment (sfc-gh-yixie) [inline CHANGELOG.md:19]: Are these two lines for the same logic branch? If so please consider merging them.

**Review by sfc-gh-yixie — COMMENTED** [review]
Code looks good. One comment on the changelog.

File: CHANGELOG.md (hunk 3 above)
Comment (sfc-gh-aalam) [reply CHANGELOG.md:19]: sure. I'll merge them

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**